### DISCUSSION TITLE
In Keycloak 21, is LDAP federation supposed to work with map storage/CockroachDB?

### DISCUSSION THREAD

**Discussion by holm-informatik** [keycloak/keycloak#17406] [Archive]
I'm trying to set up a test environment with Keycloak 21, using map storage and CockroachDB. The LDAP connection is fine but, other than that, I can't get any data from LDAP. In fact, Keycloak does not even contact the LDAP server. If I try to sync all the users I get a "Could not sync users: 'HTTP 404 Not Found'" error.

Am I doing something wrong or is this configuration a no go?

Thank You in advance.

**Comment by vramik** [comment]
Hello @holm-informatik, legacy LDAP federation isn't supposed to work together with map store. Legacy store and map store are not compatible. 

There is a plan to introduce ldap implementation of the new map storage. There has been done some work already and you can watch the progress there: https://github.com/keycloak/keycloak/tree/main/model/map-ldap

To be able to use combination of multiple map store implementations withing one area (say LDAP and JPA- cockcoach for realm store) there needs to implmented functionality called tree store: https://github.com/keycloak/keycloak/issues/9742

**Reply by holm-informatik** [reply]
As I suspected... 

Thank You for clarifying!
